### PR TITLE
dcox chapter 6 exercises

### PR DESCRIPTION
**Pull request by codavex**

### CODE REVIEW

File: chapter06/dcox/ex6_1 [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+    else {
+      say "Sorry, I don't know " . $_ . "'s surname";
+    }
+  }
+  else {
+    say "Sorry, I don't recognise the name " . $_;
+  }
+}
+
```
Comment (JPeace2) [inline chapter06/dcox/ex6_1:40]: This is personal preference but I think you've used $_ too many times. You should use a loop variable instead.

File: chapter06/dcox/ex6_1 [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+              'bamm-bamm' => 'rubble',
+              'mr' => 'slate',
+              'arnold' => undef,
+              'joe' => 'rockhead',
+              'pearl' => 'slaghoople',
+              'great' => 'gazoo',
+              'tex' => 'hardrock',
+              'sam' => 'slagheap',
+              );
+
```
Comment (glaKris) [inline chapter06/dcox/ex6_1:23]: props if you never had to look these up.

File: chapter06/dcox/ex6_1 (hunk 1 above)
Comment (glaKris) [inline chapter06/dcox/ex6_1:40]: +1

File: chapter06/dcox/ex6_2 [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+use feature "say";
+
+say "Enter a list of words per line. Ctrl-D to finish";
+
+my %words;
+
+while (<STDIN>) {
+  chomp($_);
+  $words{$_} += 1;
+}
```
Comment (glaKris) [inline chapter06/dcox/ex6_2:15]: good way to get around storing the words in an array first

File: chapter06/dcox/ex6_1 (hunk 2 above)
Comment (codavex) [inline chapter06/dcox/ex6_1:23]: Wikipedia is my friend. I am prop-less. :disappointed:

File: chapter06/dcox/ex6_1 (hunk 1 above)
Comment (codavex) [inline chapter06/dcox/ex6_1:40]: I thought it was fine in this instance (there were no function calls, no assignments) but I take the point

**Pull request merged**